portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,38 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.List;
+import com.google.gson.Gson;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Sehhaj!</h1>");
+  public void doGet(final HttpServletRequest request, final HttpServletResponse response) throws IOException {
+
+        // Initialise messages with three hard-coded values
+        ArrayList<String> messages = new ArrayList<String>();
+        messages.add("Lesser leather never weathered wetter weather better");
+        messages.add("Which witch is which?");
+        messages.add("Fred fed Ted bread, and Ted fed Fred bread");
+
+        // Convert to JSON format
+        String json = convertToJsonUsingGson(messages);
+
+        // Send JSON as response
+        response.setContentType("application/json;");
+        response.getWriter().println(json);
+    }
+
+    /**
+     * Converts a an ArrayList<String> instance into a JSON string using the Gson
+     * library.
+     */
+    private String convertToJsonUsingGson(ArrayList<String> data) {
```

Comment:
Two things here:
1) You don't need to state that it converts 'an ArrayList<String>' since the method signature tells us that. Comments shouldn't repeat information or state the obvious. Both the comment and the function name also omit something that could be of interest to the caller: the returned value isn't just a JSON string, it's a JSON array.

2) Both the comment and the method signature mention that the method uses the Gson library. This is what we call an 'implementation detail'. The caller shouldn't need to worry about what kind of library you use to do the conversion. The important thing is that the output is a json array.

With those things in mind I'd reword these to be something along the lines of:
```
// Converts the input to a JSON array.
private String toJsonArray(List<String> items)
```

Another thing I've done is omit the word 'convert'. This strays more into preference territory, but you want to keep function / method names as short as possible while still being descriptive, and you want to avoid repeating information. With that in mind, in this context I think a reader will understand 'convertToX' and 'toX' to mean the same thing, so the shorter form is preferable.

Reply:
Done :)